railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -244,11 +251,31 @@ def create_test_files
       end
 
       def create_tmp_files
-        build(:tmp)
+        build(:tmp) unless options[:api]
       end
 
       def create_vendor_files
-        build(:vendor)
+        build(:vendor) unless options[:api]
```

Comment:
Anso, if you have a very closed-off system, using the `bundle package --all` or `bundle --path vendor/bundle` both uses the vendor dir.